TrafficCapture/nettyWireLogging/src/test/java/org/opensearch/migrations/trafficcapture/netty/RootWireLoggingContextTest.java

```diff
@@ -0,0 +1,103 @@
+
+@Slf4j
+public class RootWireLoggingContextTest {
+
+  private static void writeMessageAndVerifyTraces(byte[] fullTrafficBytes, Consumer<EmbeddedChannel> channelWriter, boolean shouldBlock,
+                                                  List<String> expectedTraces) throws IOException {
+    try (var rootContext = new TestRootContext(true, true)) {
+      var streamManager = new TestStreamManager();
+      var offloader = new StreamChannelConnectionCaptureSerializer("Test", "c", streamManager);
+
+      EmbeddedChannel channel = new EmbeddedChannel(
+          new ConditionallyReliableLoggingHttpHandler(rootContext,
+              "n", "c", ctx -> offloader,
+              new RequestCapturePredicate(), x -> shouldBlock));
+      channelWriter.accept(channel);
+
+      // we wrote the correct data to the downstream handler/channel
+      var outputData = new SequenceInputStream(Collections.enumeration(channel.inboundMessages().stream()
+          .map(m -> new ByteArrayInputStream(consumeIntoArray((ByteBuf) m)))
+          .collect(Collectors.toList())))
+          .readAllBytes();
+      Assertions.assertArrayEquals(fullTrafficBytes, outputData);
+
+      // For a non-blocking write, close the channel to force it to flush
+      if (!shouldBlock) {
+        channel.close().awaitUninterruptibly();
+      }
+
+      Assertions.assertNotNull(streamManager.byteBufferAtomicReference.get(),
+          "This would be null if the handler didn't block until the output was written");
+
+      var trafficStream = TrafficStream.parseFrom(streamManager.byteBufferAtomicReference.get());
+      Assertions.assertTrue(trafficStream.getSubStreamCount() > 0 &&
+          trafficStream.getSubStream(0).hasRead());
+      var combinedTrafficPacketsStream =
+          new SequenceInputStream(Collections.enumeration(trafficStream.getSubStreamList().stream()
+              .filter(TrafficObservation::hasRead)
+              .map(to -> new ByteArrayInputStream(to.getRead().getData().toByteArray()))
+              .collect(Collectors.toList())));
+      Assertions.assertArrayEquals(fullTrafficBytes, combinedTrafficPacketsStream.readAllBytes());
+      Assertions.assertEquals(1, streamManager.flushCount.get());
+
+      List<SpanData> finishedSpans = rootContext.instrumentationBundle.getFinishedSpans();
+      Assertions.assertTrue(!finishedSpans.isEmpty());
+      Assertions.assertTrue(!rootContext.instrumentationBundle.getFinishedMetrics().isEmpty());
+
+      List<String> finishedSpanNames = finishedSpans.stream().map(x -> x.getName()).collect(Collectors.toList());
+      expectedTraces.forEach(trace ->
+          Assertions.assertTrue(finishedSpanNames.contains(trace), "finishedSpans does not contain " + trace)
+      );
+      if (shouldBlock) {
+        Assertions.assertTrue(finishedSpanNames.contains("blocked"), "finishedSpans does not contain 'blocked' on a blocking request");
+      }
```

Comment:
Is this block redundant?  It feels like it should be, even if expectedTraces didn't include this name, it could/should be added.